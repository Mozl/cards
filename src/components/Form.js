import React, { Component } from 'react';
import styled from 'styled-components';
import AnywhereCard from './AnywhereCard';
import LiquidCard from './LiquidCard';
import StudentCard from './StudentCard';

const StyledInput = styled.input`
  margin: 0.5rem;
  align-items: left;
`;

const CardWrapper = styled.div`
  width: 100%;
`;

class Form extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      dob: '',
      status: '',
      house: '',
      postcode: '',
      income: ''
    };
  }

  handleNameChange = event => {
    this.setState({
      name: event.target.value
    });
  };

  handleDobChange = event => {
    this.setState({
      dob: event.target.value
    });
  };

  handleStatusChange = event => {
    this.setState({
      status: event.target.value
    });
  };

  handleHouseChange = event => {
    this.setState({
      house: event.target.value
    });
  };

  handlePostcodeChange = event => {
    this.setState({
      postcode: event.target.value
    });
  };

  handleIncomeChange = event => {
    this.setState({
      income: event.target.value
    });
  };

  handleSubmit = event => {
    console.log(this.state);
    event.preventDefault();
  };

  submitUserDetails = e => {
    console.log(this.state.name);
    e.preventDefault();
    console.log(this.nameRef);

    const user = {
      name: this.state.name,
      dob: this.state.dob,
      status: this.state.status,
      house: this.state.house,
      postcode: this.state.postcode,
      income: this.state.income
    };
    this.props.addUser(user);
  };

  render() {
    return (
      <>
        <form onSubmit={this.handleSubmit}>
          <label>
            Name:
            <StyledInput
              type="text"
              name="name"
              value={this.state.name}
              onChange={this.handleNameChange}
            />
          </label>
          <label>
            Date of Birth:
            <StyledInput
              type="text"
              name="dob"
              value={this.state.dob}
              onChange={this.handleDobChange}
            />
          </label>
          <label>
            Employment Status:
            <StyledInput
              type="text"
              name="status"
              value={this.state.status}
              onChange={this.handleStatusChange}
            />
          </label>
          <label>
            House Number:
            <StyledInput
              type="text"
              name="house"
              value={this.state.house}
              onChange={this.handleHouseChange}
            />
          </label>
          <label>
            Postcode:
            <StyledInput
              type="text"
              name="postcode"
              value={this.state.postcode}
              onChange={this.handlePostcodeChange}
            />
          </label>
          <label>
            Annual Income:
            <StyledInput
              type="text"
              name="income"
              value={this.state.income}
              onChange={this.handleIncomeChange}
            />
          </label>
          <StyledInput type="submit" value="Submit" />
        </form>
        <CardWrapper>
          <AnywhereCard user={this.state} />
          <LiquidCard user={this.state} />
          <StudentCard user={this.state} />
        </CardWrapper>
      </>
    );
  }
}

export default Form;
