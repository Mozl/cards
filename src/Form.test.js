import React from 'react';
import ReactDOM from 'react-dom';
import Form from './components/Form';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Form />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly', () => {
  const tree = renderer.create(<Form />);
  expect(tree).toMatchSnapshot();
});

it('shows student card when student entered as Employment Status', () => {
  const wrapper = mount(<Form />);
  const statusField = wrapper.children().find('input.status');
  statusField.simulate('keydown', { which: 's' });
  statusField.simulate('keydown', { which: 't' });
  statusField.simulate('keydown', { which: 'u' });
  statusField.simulate('keydown', { which: 'd' });
  statusField.simulate('keydown', { which: 'e' });
  statusField.simulate('keydown', { which: 'n' });
  statusField.simulate('keydown', { which: 't' });
  expect(wrapper.prop('value').length).toEqual(7);
});

it('finds anywhere card', () => {
  const wrapper = mount(<Form />);
  expect(wrapper.find('AnywhereCard').length).toEqual(1);
});
