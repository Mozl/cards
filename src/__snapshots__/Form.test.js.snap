// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <form
    className="sc-gZMcBi rzLcK"
  >
    <div
      className="sc-gqjmRU hvolDC"
    >
      Name:
      <input
        className="sc-dnqmqq bOswFF"
        name="name"
        type="text"
      />
    </div>
    <div
      className="sc-gqjmRU hvolDC"
    >
      Date of Birth:
      <input
        className="sc-dnqmqq bOswFF"
        name="dob"
        placeholder="dd/mm/yy"
        type="text"
      />
    </div>
    <div
      className="sc-gqjmRU hvolDC"
    >
      Employment Status:
      <input
        className="status sc-dnqmqq bOswFF"
        onChange={[Function]}
        placeholder="full/part time or student"
        type="text"
        value=""
      />
    </div>
    <div
      className="sc-gqjmRU hvolDC"
    >
      House Number:
      <input
        className="sc-dnqmqq bOswFF"
        name="house"
        type="text"
      />
    </div>
    <div
      className="sc-gqjmRU hvolDC"
    >
      Postcode:
      <input
        className="sc-dnqmqq bOswFF"
        name="postcode"
        type="text"
      />
    </div>
    <div
      className="sc-gqjmRU hvolDC"
    >
      Annual Income £:
      <input
        className="sc-dnqmqq bOswFF"
        name="income"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
  </form>,
  <h1
    className="sc-jTzLTM evAbCp"
  >
    The best cards for you
  </h1>,
  <div
    className="sc-iwsKbI fzIXaw"
  >
    <div
      className="sc-bwzfXH uTxCW"
    >
      <div
        className="cardName sc-htpNat keciuZ"
        onClick={[Function]}
        selected={true}
      >
        <h2>
          Anywhere Card
        </h2>
      </div>
    </div>
  </div>,
  <h3
    className="creditTotal sc-VigVT diGRdv"
  >
    Total credit available to you: £
    0
  </h3>,
]
`;
